pandas/core/indexes/multi.py

```diff
@@ -3142,7 +3142,7 @@ def _update_indexer(
                             self._get_level_indexer(x, level=i, indexer=indexer)
                         )
                         indexers = (idxrs if indexers is None else indexers).union(
-                            idxrs
+                            idxrs, sort=False
```

Comment:
good catch.  there's an `intersection` a few lines up without a `sort=False`.  could investigate to make sure thats not hiding a problem

Reply:
False is the default value for ``intersection`` so we are good here.